### PR TITLE
new rule: no-unauthorized-global-properties

### PR DESCRIPTION
**Pull request by alxndr**
* [x] add ability to select preset property lists ("browser", "node", "both")

### CODE REVIEW

File: lib/rules/no-unauthorized-global-properties.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+var defaultAllowedProperties = [
+  "addEventListener",
+  "clearInterval",
```
Comment (alxndr) [inline lib/rules/no-unauthorized-global-properties.js]: This is not a complete list... where to get one?

Comment (abritinthebay) [inline lib/rules/no-unauthorized-global-properties.js]: Node at least has these: https://nodejs.org/api/globals.html

Comment (abritinthebay) [inline lib/rules/no-unauthorized-global-properties.js]: Technically anything is global that's on the `window` object in the browser.

So all the properties and methods here: https://developer.mozilla.org/en-US/docs/Web/API/Window

File: lib/rules/no-unauthorized-global-properties.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+	"parseInt",
+	"undefined",
+	"unescape"
+];
+
+var nodeGlobalProperties = [
+	"__dirname",
+	"__filename",
+	"clearImmediate",
+	"clearInterval",
```
Comment (abritinthebay) [inline lib/rules/no-unauthorized-global-properties.js]: intervals and timeout functions are node? Surely those are just standard (same with console)?

**Pull request merged**